src/doppler/main.go

```diff
@@ -163,11 +176,12 @@ func StartHeartbeats(localIp string, ttl time.Duration, config *config.Config, l
 		return
 	}
 
-	adapter := StoreAdapterProvider(config.EtcdUrls, config.EtcdMaxConcurrentRequests)
-	adapter.Connect()
+	if storeAdapter == nil {
```

Comment:
sorry that reply you so late.
In working state, it's impossible,  but I think `StartHeartbeats` is a public function, want to make sure `storeAdapter` isn't nil.